tools/matc/src/matc/MaterialCompiler.cpp

```diff
@@ -441,4 +453,49 @@ bool MaterialCompiler::parseMaterial(const char* buffer, size_t size,
     return true;
 }
 
+bool MaterialCompiler::compileRawShader(const char* glsl, size_t size,
+        Config::Output* output, const char* ext) const noexcept {
+    using namespace glslang;
+    using namespace filament::backend;
+    using SpirvBlob = std::vector<uint32_t>;
+
+    const EShLanguage shLang = !strcmp(ext, "vs") ? EShLangVertex : EShLangFragment;
+
+    TProgram program;
+    TShader tShader(shLang);
+    tShader.setStrings(&glsl, 1);
+
+    const int version = 110;
+    EShMessages msg = EShMessages::EShMsgDefault;
+    msg = (EShMessages) (EShMessages::EShMsgVulkanRules | EShMessages::EShMsgSpvRules);
+
+    tShader.setAutoMapBindings(true);
+
+    bool ok = tShader.parse(&DefaultTBuiltInResource, version, false, msg);
+    if (!ok) {
+        std::cerr << "ERROR: Unable to parse " << ext << ":" << std::endl;
+        std::cerr << tShader.getInfoLog() << std::endl;
+        return false;
+    }
```

Comment:
I think we need to make sure we link the program, just as we do for normal materials:
https://github.com/google/filament/blob/9f68daeb8f18b8d13e66105a009e59644531a8a2/libs/filamat/src/GLSLPostProcessor.cpp#L202-L209

Reply:
Good catch, fixed.